api/proto/teleport/legacy/types/events/events.proto

```diff
@@ -214,6 +214,38 @@ message OktaAssignmentMetadata {
   string EndingStatus = 4 [(gogoproto.jsontag) = "ending_status,omitempty"];
 }
 
+// AccessListMemberMetadata contains common metadata for access list member resource events.
+message AccessListMemberMetadata {
+  // joined_on is the date that the member joined.
+  google.protobuf.Timestamp joined_on = 1 [
+    (gogoproto.stdtime) = true,
+    (gogoproto.nullable) = false,
+    (gogoproto.jsontag) = "session_start,omitempty"
+  ];
+
+  // removed_on is the date that the access list member was removed.
+  google.protobuf.Timestamp removed_on = 2 [
```

Comment:
Should this be a oneof? Or is it possible that both `joined_on` and `removed_on` are set?

Reply:
I'm a little leery of using `oneof` for these events. It's not used this way, but there's no reason you couldn't populate both here, so I'd rather keep this the way it is.

Reply:
Sounds good. Maybe a comment to say that `removed_on` is only populated when a member is removed would help. (Though this is a moot point if you end up creating separate events for add and remove, as you don't need to stick fields in there that don't apply to the event)

Reply:
Despite separating the events, I did leave this `removed_on` field just to avoid having to have another `Metadata` message, but I can separate this if you feel strongly.